std/algorithm/comparison.d

```diff
@@ -1085,15 +1085,52 @@ size_t levenshteinDistance(alias equals = "a == b", Range1, Range2)
     (Range1 s, Range2 t)
     if (isForwardRange!(Range1) && isForwardRange!(Range2))
 {
-    Levenshtein!(Range1, binaryFun!(equals), size_t) lev;
+    alias comp = binaryFun!(equals);
+
+    while (!s.empty() && !t.empty() && comp(s.front, t.front))
+    {
+        s.popFront();
+        t.popFront();
+    }
+
+    static if (isBidirectionalRange!(Range1) && isBidirectionalRange!(Range2))
+    {
+        while (!s.empty() && !t.empty() && comp(s.back, t.back))
+        {
+            s.popBack();
+            t.popBack();
+        }
+    }
+
+    if (s.empty())
+    {
+        return walkLength(t);
+    }
+
+    if (t.empty())
+    {
+        return walkLength(s);
+    }
+
     auto slen = walkLength(s.save);
     auto tlen = walkLength(t.save);
+
+    if (slen == 1 && tlen == 1)
+    {
+        if (comp(s.front, t.front))
```

Comment:
My mistake, I didn't notice that the true case will produce 0, instead of vice-versa. It should be `return cast(int)!comp(s.front, t.front)`.